orderapi/src/main/java/fi/hel/verkkokauppa/order/service/order/OrderService.java

```diff
@@ -391,24 +382,41 @@ public Order getLatestOrderWithSubscriptionId(String subscriptionId) {
         return lastOrder.map(orderAggregateDto -> findById(orderAggregateDto.getOrder().getOrderId())).orElse(null);
     }
 
-    public JSONObject saveOrderMessageHistory(OrderMessage message){
-        try {
-            String request = historyUtil.toString(historyFactory.fromOrderMessage(message));
-            return restServiceClient.makePostCall(serviceUrl.getHistoryServiceUrl() + "/history/create",request);
-        } catch (Exception e) {
-            log.error("saveOrderMessageHistory processing error: " + e.getMessage());
+    public FlowStepDto saveFlowStepsByOrderId(String orderId, FlowStepDto flowStepDto) {
+        if (flowStepDto == null) {
+            throw new RuntimeException("Failed to save flow steps, no data provided to save");
         }
-        return null;
-    }
-
-    public JSONObject savePaymentMessageHistory(PaymentMessage message){
-        try {
-            String request = historyUtil.toString(historyFactory.fromPaymentMessage(message));
-            return restServiceClient.makePostCall(serviceUrl.getHistoryServiceUrl() + "/history/create",request);
-        } catch (Exception e) {
-            log.info("savePaymentMessageHistory processing error: " + e.getMessage());
+        orderRepository.findById(orderId)
+                .orElseThrow(() -> new CommonApiException(
+                        HttpStatus.NOT_FOUND,
+                        new Error("order-not-found", "order with id [" + orderId + "] not found")
+                ));
+
+        // Update existing flow step or create new by orderId
+        Optional<FlowStep> existingFlowStep = flowStepRepository.findByOrderId(orderId).stream().findFirst();
```

Comment:
Could you make new service to handle this logic?

Reply:
Yes, I separated the flow step related methods to new "FlowStepService" class.